specification/network/resource-manager/Microsoft.Network/stable/2019-11-01/networkWatcher.json

```diff
@@ -1525,1535 +1525,2590 @@
-        }
-      },
-      "allOf": [
-        {
-          "$ref": "#/definitions/PacketCaptureParameters"
-        }
+    "ConnectivityParameters": {
+      "description": "Parameters that determine how the connectivity check will be performed.",
+      "required": [
+        "source",
+        "destination"
       ],
-      "description": "The properties of a packet capture session."
-    },
-    "PacketCaptureQueryStatusResult": {
       "properties": {
-        "name": {
-          "type": "string",
-          "description": "The name of the packet capture resource."
-        },
-        "id": {
-          "type": "string",
-          "description": "The ID of the packet capture resource."
+        "source": {
+          "$ref": "#/definitions/ConnectivitySource",
+          "description": "The source of the connection."
         },
-        "captureStartTime": {
-          "type": "string",
-          "format": "date-time",
-          "description": "The start time of the packet capture session."
+        "destination": {
+          "$ref": "#/definitions/ConnectivityDestination",
+          "description": "The destination of connection."
         },
-        "packetCaptureStatus": {
+        "protocol": {
           "type": "string",
+          "description": "Network protocol.",
           "enum": [
-            "NotStarted",
-            "Running",
-            "Stopped",
-            "Error",
-            "Unknown"
+            "Tcp",
+            "Http",
+            "Https",
+            "Icmp"
           ],
           "x-ms-enum": {
-            "name": "PcStatus",
+            "name": "Protocol",
             "modelAsString": true
-          },
-          "description": "The status of the packet capture session."
+          }
         },
-        "stopReason": {
-          "type": "string",
-          "description": "The reason the current packet capture session was stopped."
+        "protocolConfiguration": {
+          "$ref": "#/definitions/ProtocolConfiguration",
+          "description": "Configuration of the protocol."
         },
-        "packetCaptureError": {
-          "type": "array",
-          "description": "List of errors of packet capture session.",
-          "items": {
-            "type": "string",
-            "enum": [
-              "InternalError",
-              "AgentStopped",
-              "CaptureFailed",
-              "LocalFileFailed",
-              "StorageFailed"
-            ],
-            "x-ms-enum": {
-              "name": "PcError",
-              "modelAsString": true
-            }
-          }
+        "preferredIPVersion": {
+          "$ref": "./network.json#/definitions/IPVersion",
+          "description": "Preferred IP version of the connection."
         }
-      },
-      "description": "Status of packet capture session."
+      }
     },
-    "TroubleshootingParameters": {
-      "description": "Parameters that define the resource to troubl
```

Comment:
The snippets on these comments are not showing correctly in conversation view.  You'll see them in the correct place in the files view on networkWatcher.json.  I also commented on one of Kris's previous questions

Reply:
Chris, the problem is that we have 2 parameters sets in connection monitor object. They don't inetersect with each other. So, I could specify requred parameters inside of sets, but not in the whole object...

Reply:
Is this where "connectionMonitorType" comes in?  If connection monitors can have two different schemas swagger has support for polymorphism (and its used throughout the repo).  Take a look at docs about ["discriminator"](https://swagger.io/specification/v2/) here and examples of "discriminator" and ["x-ms-discriminator-value"](https://github.com/Azure/autorest/blob/master/docs/extensions/readme.md#x-ms-discriminator-value) throughout the repo.

Otherwise, how does the user know what values to provide to the connection monitor when they are creating it via SDKs?

Reply:
If I understand correctly, we need to have 'discriminator' parameter which will allow to differentiate one parameter set from another. Unfortunately, it's nor the case:( "ConnectionMonitorType" which you mentioned is a readonly property which is returned only in the result of GET call. Customer can't set it.

We are going to document usage cases so customers would know which set they should use.
Actually, Chris, connection monitor is an old product: https://github.com/Azure/azure-rest-api-specs/blob/master/specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/networkWatcherConnectionMonitorV1.json

Some time ago it was decided to make it more powerfull. The idea was to keep old connectionMonitor resource. Add a new set of properties to it (so, now connection monitor is a superset of old and new properties). API version defines which connectionMonitor (V1 or V2) should be created.

ConnectionMonitor V2 is going to be in public preview in GA. For at least 6 months we are going to support both V1 and V2 properties. Then, we allow customers to convert V1 to V2. At some point of time all V1 monitors will be converted into V2 and, then, we will deprecate V1.

We need both sets of properties in swagger now to not break customer experience in powershell and CLI where, depending on parameter set, we will call new API version (and create connectionMonitor V2) or old API version (connectionMonitor V1)

It was already discuss with your team several months ago. And we got an approval. If you feel you need more details, let's sync up and discuss...